test/floor/allocators/NFTXAllocator.js

```diff
@@ -0,0 +1,647 @@
+   */
+
+  it("Should be able to set and remove NFTX vault mappings", async function () {
+    // Try and set a null address token
+    expect(allocator.setStakingToken(NULL_ADDRESS, rewardToken.address, 1, true)).to.be.revertedWith('Cannot set vault for NULL token');
+
+    // Set a vault against a token
+    await allocator.setStakingToken(TEST_ADDRESS_1, rewardToken.address, 1, true);
+
+    mapping = await allocator.stakingTokenInfo(TEST_ADDRESS_1);
+    expect(mapping.rewardToken).to.equal(rewardToken.address);
+    expect(mapping.vaultId).to.equal(1);
+    expect(mapping.isLiquidityPool).to.equal(true);
+    expect(mapping.exists).to.equal(true);
+
+    // Update the vault mapping
+    await allocator.setStakingToken(TEST_ADDRESS_1, rewardToken.address, 2, false);
+
+    mapping = await allocator.stakingTokenInfo(TEST_ADDRESS_1);
+    expect(mapping.rewardToken).to.equal(rewardToken.address);
+    expect(mapping.vaultId).to.equal(2);
+    expect(mapping.isLiquidityPool).to.equal(false);
+    expect(mapping.exists).to.equal(true);
+
+    // Delete the vault mapping
+    allocator.removeStakingToken(TEST_ADDRESS_1);
+
+    mapping = await allocator.stakingTokenInfo(TEST_ADDRESS_1);
+    expect(mapping.rewardToken).to.equal(NULL_ADDRESS);
+    expect(mapping.vaultId).to.equal(0);
+    expect(mapping.isLiquidityPool).to.equal(false);
+    expect(mapping.exists).to.equal(false);
+  });
+
+
+  /**
+   * Confirm that our harvest function is reverted as we're unable to specify
+   * an amount to request as a parameter, but still need to adhere to the
+   * interface.
+   */
+
+  it("Should not allow for harvest function call", async function () {
+    expect(
+      allocator.harvest(TEST_ADDRESS_1, 100)
+    ).to.be.revertedWith('Method is deprecated in favour of harvestAll(address _token)');
+  });
+
+
+  /**
+   * Confirm that we can run a harvestAll command against a known vault.
+   */
+
+  it("Should be able to harvest all rewards against a token", async function () {
+    // Approve our reward token in the treasury
+    await treasury.enable(2, rewardToken.address, bondingCalculator.address);  // RESERVETOKEN
+    await treasury.enable(0, allocator.address, bondingCalculator.address);    // RESERVEDEPOSITOR
+
+    // Set up an inventory and liquidity pool vaults
+    await allocator.setStakingToken(TEST_ADDRESS_1, rewardToken.address, 123, false);  // Inventory
+    await allocator.setStakingToken(TEST_ADDRESS_2, rewardToken.address, 321, true);   // Liquidity
+
+    // Confirm that an invalid mapping with no vault ID will be prevented
+    expect(allocator.harvestAll(TEST_ADDRESS_3)).to.be.revertedWith('Unsupported token');
+
+    // Confirm that we cannot claim rewards from an inventory pool
+    expect(allocator.harvestAll(TEST_ADDRESS_1)).to.be.revertedWith('Must be liquidity staking token');
+
+    // Trigger our rewards to be claimed against the mapped vault ID
+    await allocator.harvestAll(TEST_ADDRESS_2);
```

Comment:
Getting this error

`Error: Transaction reverted: function call to a non-contract account`